src/Gaffer/HiddenFilePathFilter.cpp

```diff
@@ -0,0 +1,117 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2022, Hypothetical Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/FileSystemPath.h"
+#include "Gaffer/HiddenFilePathFilter.h"
+
+#include "Gaffer/Path.h"
+
+#include "boost/bind.hpp"
+
+#ifdef _MSC_VER
+
+#include <windows.h>
+
+#endif
+
+using namespace Gaffer;
+
+IE_CORE_DEFINERUNTIMETYPED( HiddenFilePathFilter );
+
+HiddenFilePathFilter::HiddenFilePathFilter( IECore::CompoundDataPtr userData )
+	:	PathFilter( userData ), m_inverted( false )
+{
+}
+
+HiddenFilePathFilter::~HiddenFilePathFilter()
+{
+}
+
+void HiddenFilePathFilter::setInverted( bool inverted )
+{
+	if( inverted == m_inverted )
+	{
+		return;
+	}
+	m_inverted = inverted;
+	changedSignal()( this );
+}
+
+bool HiddenFilePathFilter::getInverted() const
+{
+	return m_inverted;
+}
+
+void HiddenFilePathFilter::doFilter( std::vector<PathPtr> &paths, const IECore::Canceller *canceller ) const
+{
+	paths.erase(
+		std::remove_if(
+			paths.begin(),
+			paths.end(),
+			boost::bind( &HiddenFilePathFilter::remove, this, ::_1 )
```

Comment:
In my cherry-pick PR I used a lambda here, since `bind` is so 1998.

Reply:
Thanks - 90's trends are making a comeback, but I'm happy to leave `bind` in the past.